src/component/Todos.js

```diff
@@ -0,0 +1,190 @@
+import React from 'react'
+import { useState, useEffect } from 'react'
+import moment from "moment";
+import { FiPlusCircle } from "react-icons/fi";
+import Navbar from './Navbar'
+import TodoItem from './TodoItem';
+import AddTodoModal from './AddTodoModal'
+import DeleteTodoModal from './DeleteTodoModal';
+const Todos = () => {
+  const [todoList, setTodoList] = useState([]);
+  const [isShowTodoPopup, setIsShowTodoPopup] = useState(false);
+  const [isShowDeleteModal, setIsShowDeleteModal] = useState(false); // State for showing/hiding delete todo modal
+  const [todoInputText, setTodoInputText] = useState("");
+  const [inputError, setInputError] = useState(false);
+  const [dateError,setDateError] = useState(false)
+  const [deleteTodoId, setDeleteTodoId] = useState(null);
+  const [editTodoId, setEditTodoId] = useState(null);
+  const [currentDateAndTime, setCurrentDateAndTime] = useState(moment().format('YYYY-MM-DDTHH:mm'))   // State for current date and time
+  const [time, setTime] = useState(moment().format('YYYY-MM-DDTHH:mm'))
+
+ // Function to show add todo modal
+  const showTodoPopup = () => {
+    setIsShowTodoPopup(true);
+    setTime(moment().format('YYYY-MM-DDTHH:mm'))  // Reset time when showing popup
+  }
+   // Function to hide add todo modal
+  const hideTodoPopup = () => {
+    setIsShowTodoPopup(false);
+    setTime(moment().format('YYYY-MM-DDTHH:mm'))
+    setTodoInputText("")
+  }
+
+    // Function to show delete todo modal
+  const showDeletePopupModal = (id) => {
```

Comment:
You can also delete popup handle in one function don't create two seprate function for this.